addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -609,11 +609,20 @@ def _import_fill_invoice_line_values(self, tree, xpath_dict, invoice_line, qty_f
         if billed_qty * price_unit != 0 and price_subtotal is not None:
             discount = 100 * (1 - (price_subtotal - amount_fixed_taxes) / (billed_qty * price_unit))
 
-        # Sometimes, the xml received is very bad: unit price = 0, qty = 1, but price_subtotal = -200
+        # Sometimes, the xml received is very bad; e.g.:
+        #   * unit price = 0, qty = 0, but price_subtotal = -200
+        #   * unit price = 0, qty = 1, but price_subtotal = -200
+        #   * unit price = 1, qty = 0, but price_subtotal = -200
         # for instance, when filling a down payment as an invoice line. The equation in the docstring is not
         # respected, and the result will not be correct, so we just follow the simple rule below:
-        if net_price_unit == 0 and price_subtotal != net_price_unit * (billed_qty / basis_qty) - allow_charge_amount:
-            price_unit = price_subtotal / (billed_qty or 1)
+        if net_price_unit is not None and price_subtotal != net_price_unit * (billed_qty / basis_qty) - allow_charge_amount:
+            if net_price_unit == 0 and billed_qty == 0:
+                quantity = 1
+                price_unit = price_subtotal
+            elif net_price_unit == 0:
+                price_unit = price_subtotal / billed_qty
+            elif billed_qty == 0:
+                quantity = price_subtotal / price_unit
```

Comment:
Plz add a test for that

Reply:
I added a test covering all 3 cases of billed_qty and/or net_price_unit being 0